src/services/ExposureNotificationService/ExposureNotificationService.ts

```diff
@@ -175,25 +174,10 @@ export class ExposureNotificationService {
 
     const lastCheckPeriod = periodSinceEpoch(lastFetchDate || addDays(runningDate, -14), hoursPerPeriod);
     let runningPeriod = periodSinceEpoch(runningDate, hoursPerPeriod);
-    while (runningPeriod > lastCheckPeriod) {
-      const zipFile = await this.backendInterface.retrieveDiagnosisKeys(runningPeriod);
-      if (zipFile) {
-        const components = zipFile.split('/');
-        components.pop();
-        components.push('keys-export');
-        const targetDir = components.join('/');
-
-        try {
-          yield await unzip(zipFile, targetDir);
-        } catch (err) {
-          yield null;
-        }
-      } else {
-        yield null;
-      }
 
-      // yield await this.backendInterface.retrieveDiagnosisKeys(runningPeriod);
-      runningPeriod -= hoursPerPeriod;
+    while (runningPeriod > lastCheckPeriod) {
+      yield await this.backendInterface.retrieveDiagnosisKeys(runningPeriod).catch(() => null);
+      runningPeriod -= 1;
```

Comment:
We try to retrieve 13 days history + 1 current day. Although the server will return fail for current day but the catch helps.